### PR TITLE
ECT-726 hotfix metadata

### PR DESCRIPTION
**Pull request by chris-clifford**
This fixes an issue where the metadata was not updated with the per-request configuration.

### CODE REVIEW

File: lib/saml_idp.rb [hunk 1]
```diff
@@ -9,17 +9,26 @@ module SamlIdp
...
+    end
   end
 
   def self.configure
     yield config
   end
 
   def self.metadata
-    @metadata ||= MetadataBuilder.new(config)
+    MetadataBuilder.new(config)
```
Comment (amirkarimi) [inline lib/saml_idp.rb:27]: Not sure how performant `MetadataBuilder` is but if it's heavy we can use a `RequestStore` variable and make it a request-level singleton.

File: lib/saml_idp.rb [hunk 2]
```diff
@@ -9,17 +9,26 @@ module SamlIdp
...
+  end
+
+  def self.request_config=(config)
```
Comment (amirkarimi) [inline lib/saml_idp.rb:30]: Where is this used?

File: lib/saml_idp.rb (hunk 1 above)
Comment (amirkarimi) [reply lib/saml_idp.rb:27]: Just took a look at MetadataBuilder seems like the constructor is just setting an instance variable so we're good.

Comment (chris-clifford) [reply lib/saml_idp.rb:27]: Yeah I thought about that, but I wanted to make minimal changes and I thought that metadata is something that is not requested very frequently. Usually on requested during setup.

Do you think I should change to request store though?

Like this?
```ruby
  def self.metadata
    RequestStore.store[:metadata] ||= MetadataBuilder.new(config)
  end
```

File: lib/saml_idp.rb (hunk 2 above)
Comment (chris-clifford) [reply lib/saml_idp.rb:30]: This sets the per request config in host application.

File: lib/saml_idp.rb (hunk 1 above)
Comment (amirkarimi) [reply lib/saml_idp.rb:23]: No, let's keep it as it is. We should monitor the performance and if it was problematic we can update it.

**Pull request merged**